staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -86,6 +100,10 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.Version
 
 // note that callAttrMutatingHook updates attr
 func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta1.Webhook, attr *generic.VersionedAttributes, o admission.ObjectInterfaces) error {
+	reinvokeCtx := attr.GetReinvocationContext()
+	if h.ReinvocationPolicy != nil && *h.ReinvocationPolicy == v1beta1.IfNeededReinvocationPolicy {
+		defer reinvokeCtx.AddReinvocableWebhookToPreviouslyInvoked(h.Name)
+	}
```

Comment:
move this whole block outside of `callAttrMutatingHook` back into Dispatch in order to make the logic more localized.

Reply:
This works out nice. It eliminates the need to defer the call as well.